recipes/nv-codec-headers/all/conanfile.py

```diff
@@ -0,0 +1,50 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+import os
+
+required_conan_version = ">=1.35.0"
+
+class FFNvEncHeaders(ConanFile):
+    name = "nv-codec-headers"
+    description = "FFmpeg version of headers required to interface with Nvidia's codec APIs"
+    topics = ("ffmpeg", "video", "nvidia", "headers")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/FFmpeg/nv-codec-headers"
+    license = "MIT"
+
+    _autotools = None
+
+    def package_id(self):
+        self.info.header_only()
+
+    def build_requirements(self):
+        if tools.os_info.is_windows:
+            if "CONAN_MAKE_PROGRAM" not in os.environ:
+                self.build_requires("make/4.2.1")
+
+    def source(self):
```

Comment:
Use `self.settings_build` (if available) for cross building support.
```suggestion
    _autotools = None

    @property
    def _settings_build(self):
        return getattr(self, "settings_build", self.settings)

    def build_requirements(self):
        if self._settings_build.os == "Windows":
            if "CONAN_MAKE_PROGRAM" not in os.environ:
                self.build_requires("make/4.2.1")

    def package_id(self):
        self.info.header_only()

    def source(self):
```

Reply:
this is a header only library, it has no settings

Reply:
header-only libraries can have a settings object.
As long as you clear it in `package_id`.